### PR TITLE
Fixes/do not override timestamps

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Roll back my previous changes, document the `override_timestamps` debacle a bit more, and introduce a new `override_timestamps` virtual attribute to the `Status` model used to ignore the `created_at` field when generating the snowflake ID and use the current timestamp instead.

TL;DR: `created_at` becomes the published date, `id` can optionally not be based on `created_at`

### CODE REVIEW

File: app/models/status.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #  text                   :text             default(""), not null
 #  created_at             :datetime         not null
 #  updated_at             :datetime         not null
+#  published_at           :datetime
```
[the remark this answers is not in the document]
Comment (ThibG) [reply app/models/status.rb]: Yeah that would probably be better, but I'm not sure how to do that.
Also, I'm still bothered by older statuses within realtime window being inserted in the middle of pages…

**Comment by ThibG** [comment]
I believe I have finally managed to implement it without introducing new issues :weary:

**Review by Gargron — APPROVED** [review]

**Comment by chandrn7** [comment]
@ClearlyClaire What exactly does adding back `override_timestamps` do/what was the motivation to rollback #7331? It's still a bit unclear to me. I see `override_timestamps` is set to `true` in `app/workers/activitypub/processing_worker.rb ` and `app/workers/processing_worker.rb `. What's the reason for that?

**Comment by ClearlyClaire** [comment]
@chandrn7 oof, that is an old change you've excavated here

as far as I understand, there are two places where the status' creation time matters:
- its `created_at`, which is what is displayed to the end-user
- its `id`, which determines in which order toots are displayed, affects pagination etc.

basically, `override_timestamps` makes it so the `id` is based on the time the status is processed, whereas otherwise it's based on the time it was reportedly created at

basically, this means:
- toots received through direct federation are inserted in timelines based on the time they were received, so you don't end up missing them
- toots fetched otherwise are inserted into timelines based on their reported creation date, and are distributed to clients if they're recent enough

**Comment by chandrn7** [comment]
@ClearlyClaire thanks, that helps.

What's the distinction between direct federation and "toots fetched otherwise"? And why do they have different behaviors?

**Comment by ClearlyClaire** [comment]
Direct federation is anything received directly in an inbox. Toots fetched otherwise are those fetched because they were boosted or replied to by a follower, fetched by resolving a thread, or manually fetched by a local user pasting their URL in the search box.

The main reason is that we want to avoid timelines being in anything else than chronological order, but an exception is made for direct federation if the time delta is relatively low to reduce the issues potentially caused by time skew, federation delay, or temporary errors.

**Comment by Gargron** [comment]
For example, if a previously unknown, old post has been fetched using manual URL search, we do not want it to pop up at the top of the followers' home feeds, because that would be unexpected. But if a "new" post arrives through the inbox with a 2 hour delay due to connection issues, we do want it to pop up at the top of followers' home feeds.
